cmake/FindPostgreSQL.cmake

```diff
@@ -135,6 +135,12 @@ if(NOT DEFINED PG_CONFIG)
             string(APPEND extra_configure_args " --enable-debug --enable-cassert")
         endif()
 
+        if(BUILD_TYPE STREQUAL "RelWithDebInfo")
+            set(ENV{CFLAGS} "$ENV{CFLAGS} -O2")
+        elseif(BUILD_TYPE STREQUAL "Release")
+            set(ENV{CFLAGS} "$ENV{CFLAGS} -O2")
```

Comment:
I suspect the correct variables to use here would be 
`CMAKE_C_FLAGS_RELEASE` and `CMAKE_C_FLAGS_RELWITHDEBINFO` instead of -O2